Classes/Utils/ProcessedFilesHandler.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Netresearch\RteCKEditorImage\Utils;
+
+use Symfony\Component\Process\Process;
+use TYPO3\CMS\Core\Resource\File;
+use TYPO3\CMS\Core\Resource\ProcessedFile;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Service\ImageService;
+
+class ProcessedFilesHandler
+{
+    /**
+     * Create a processed variant of a file based on the given configuration.
+     * Returns the processedFile or false if the file could not be created
+     *
+     * Example for the image configuration:
+     *   $imageConfiguration = [
+     *     'width' => '200c',
+     *     'height' => '200c',
+     *   ];
+     *
+     * @param File $file The file object
+     * @param array $imageConfiguration The image configuration
+     * @return ProcessedFile|bool
+     */
+    public function createProcessedFile(File $file, array $imageConfiguration): ProcessedFile|bool
```

Comment:
This method should always return `File`, in case of error it should throw an `exception`